include/knowrob/PropertyTree.h

```diff
@@ -26,7 +26,18 @@ namespace knowrob {
 		 * Load a reasoner configuration from a property tree.
 		 * @param ptree a property tree.
 		 */
-		explicit PropertyTree(const boost::property_tree::ptree *ptree);
+		explicit PropertyTree(std::shared_ptr<const boost::property_tree::ptree> ptree);
+
+		/**
+		 * Load a reasoner configuration from a JSON string.
+		 * @param json_str a JSON string.
+		 */
+		explicit PropertyTree(const std::string_view &json_str);
```

Comment:
minor: You should not use by reference for string_view, always pass string_view by value

@see https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/